net/src/udp/mod.rs

```diff
@@ -3,17 +3,29 @@
 
 //! UDP header type and logic.
 
+pub mod checksum;
 pub mod port;
 
-use crate::parse::{DeParse, DeParseError, LengthError, Parse, ParseError};
+use crate::parse::{DeParse, DeParseError, LengthError, Parse, ParseError, ParsePayload, Reader};
 use crate::udp::port::UdpPort;
+use crate::vxlan::Vxlan;
 use etherparse::UdpHeader;
 use std::num::NonZero;
+use tracing::debug;
 
 /// A UDP header.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Udp(UdpHeader);
 
+/// A UDP encapsulation.
+///
+/// At this point we only support VXLAN, but Geneve and others can be added as needed.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Encap {
```

Comment:
I think I would rename this to UdpEncap since it seems to be specific of UDP only.... just in case we have to support other non-udp encapsulations (say GRE), or 6to4's and the like.

Reply:
Now, I am seeing that Packet has a member encap: Option<Encap>. An alternative to rename it, and unless you nest it in Udp (see comments in packet.rs), would be to move this definition out of the udp module so that it can include a bunch of other encapsulations.

Reply:
This is fair.  Will rename.  TCP encap is rare but does exist